### DISCUSSION TITLE
How to customize the Theme of the Sliders?

### DISCUSSION THREAD

**Discussion by saif-ellafi** [hm21/pro_image_editor#280] [Q&A] [answered]
Hello,

I can't seem to be able to change the colors of the slider inside the text editor (i.e. font scale, or line width sliders).

![image](https://github.com/user-attachments/assets/b8f4795a-7ba7-443a-8273-45f319c4309e)

Tried setting this up without any success:

```dart
theme: ThemeData(
    sliderTheme: SliderThemeData(
      activeTrackColor: Colors.pink,
      inactiveTrackColor: Colors.grey,
    )
  ),
```

Any ideas welcome!

**Comment by hm21** [comment, marked as answer]
The design customization options within the `SliderThemeData` are somewhat limited, as there are a lot of configurations available for customizing a slider. Therefore, I decided it would make more sense to allow users to create their own sliders with `customWidgets`. Below is an example how you can replace the existing slider with your own implementation:

```dart
configs: ProImageEditorConfigs(
  customWidgets: ImageEditorCustomWidgets(
    textEditor: CustomWidgetsTextEditor(
      sliderFontSize:
          (editorState, rebuildStream, value, onChanged, onChangeEnd) {
        return ReactiveCustomWidget(
          stream: rebuildStream,
          builder: (_) => Slider(
            onChanged: onChanged,
            onChangeEnd: onChangeEnd,
            value: value,
            activeColor: Colors.blue.shade200,
          ),
        );
      },
    ),
  ),
),
```

**Comment by saif-ellafi** [comment]
Ohh truly wonderful, thank you, I also think I was misplacing the configs, because this is was part of a custom widget opening the text editor page manually, so the bigger scope image editor config was not applying the theme in this custom TextEditor page. In case it helps anyone:

```dart
    editorKey.currentState?.openPage(
      TextEditor(
        key: key,
        layer: newLayer,
        theme: ThemeData(),
        configs: ProImageEditorConfigs(
          customWidgets: ImageEditorCustomWidgets(
            textEditor: CustomWidgetsTextEditor(
              sliderFontSize:
                  (editorState, rebuildStream, value, onChanged, onChangeEnd) {
                return ReactiveCustomWidget(
                  stream: rebuildStream,
                  builder: (_) => Slider(
                    onChanged: onChanged,
                    onChangeEnd: onChangeEnd,
                    value: value,
                    activeColor: Colors.red,
                  ),
                );
              },
            ),
          ),
        ),
        callbacks: ProImageEditorCallbacks(
          textEditorCallbacks: TextEditorCallbacks(
            onDone: () {
              newLayer.text = key.currentState!.textCtrl.text;
            },
          )
        ),
      )
    );
```

**Comment by saif-ellafi** [comment]
Btw I do not know if closing discussions is the right thing to do, feel free to reopen.

Best regards

**Reply by hm21** [reply]
Normally, I leave Q&A discussions open because, by default, GitHub filters discussions as “open” when users visit the discussions tab. Users who are not very familiar with GitHub may not know how to change the filter and, as a result, might miss the closed discussions. I will reopen this Q&A cuz I believe it might be helpful for other users to see it directly.

Best regards
